### PR TITLE
fix(masthead): react rtl overflow

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

[Masthead RTL] - issue with overflow in react #6750

### Description

Problem: 
Masthead overflow functionality was not working in RTL. 

Solution:
Detect when the page is RTL and swap the overflow logic.

overflow rtl

### Changelog

**Changes**

- add overflow logic for `paginateRight()`, `paginateLeft()`, and for the tabbing in rtl

### CODE REVIEW

File: packages/react/src/components/Masthead/HeaderNavContainer.js [hunk 1]
```diff
@@ -30,75 +30,140 @@ const HeaderNavContainer = ({ children }) => {
   const [io, setIO] = useState(null);
   const [position, setPosition] = useState(0);
   const buttonSize = 48; // 40px(width) + 8px(gradient)
+  const pageIsRTL = document.dir === 'rtl';
```
Comment (jeffchew) [inline packages/react/src/components/Masthead/HeaderNavContainer.js:33]: @annawen1 this is causing errors in NextJS

image

File: packages/react/src/components/Masthead/HeaderNavContainer.js [hunk 2]
```diff
@@ -155,25 +221,49 @@ const HeaderNavContainer = ({ children }) => {
...
-            position <=
-            buttonSize
-        ) {
-          paginateLeft();
+      if (pageIsRTL) {
+        if (event.shiftKey) {
+          if (
+            document.activeElement.parentElement.previousSibling &&
+            document.activeElement.parentElement.previousSibling.offsetLeft +
+              document.activeElement.parentElement.previousSibling.offsetWidth -
```
Comment (emyarod) [inline packages/react/src/components/Masthead/HeaderNavContainer.js:229]: just one question, would we be able to use optional chaining here and also on L238/L240?

Comment (annawen1) [reply packages/react/src/components/Masthead/HeaderNavContainer.js:229]: sure, I can add that

**Review by jeffchew — APPROVED** [review]
LGTM!

**Pull request merged**